### DISCUSSION TITLE
Dynamic Timeline Variables

### DISCUSSION THREAD

**Discussion by lobennett** [jspsych/jsPsych#1175] [Q&A] [answered]
Hello people of jsPsych,

I was wondering if there was an update that allows for dynamic access of timeline variables (through running a function in the timeline variables plugin). I have to get ratings for images from a scale, use these ratings to rank the images, which then determines the order in which the participants sees them. Each participant's order should vary based on their rating of the images. I was wondering if this is possible, and what would be a good approach to doing so -- I can also share the code I have now if that might help.

Looking forward to the responses!
Logan

**Comment by jodeleeuw** [comment, marked as answer]
Hi Logan,

Timeline variables are kind of like a shorthand for the most common use case of dynamic parameters. When your needs are more sophisticated, like they are here, it's probably more efficient to implement this using dynamic parameters without timeline variables. Or you may find it useful to combine the approaches. For example, using timeline variables to list the images that need ratings, then computing the order using the `call-function` plugin, and then using a looping timeline to go through the list of images in order.

**Reply by lobennett** [reply]
Thanks so much! I'll look into that and if I have any trouble I'll get back to you.
